### PR TITLE
feat: implement getUserGroups

### PR DESCRIPTION
**Pull request by bhudnell**
Issue #, if available:

Description of changes:
- Implemented `getUserGroups` in DynamicAuthorizationService
- Implemented `getUserGroups` in WBCGroupManagementPlugin
- Created `getUserRoles` function in UserManagementService
- Created `getUserRoles` function in CognitoUserManagementPlugin
- Added unit and integration tests for above functions

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [x] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: workbench-core/example/app/src/configs/staticRouteConfig.ts [hunk 1]
```diff
@@ -107,6 +107,15 @@ export const routesMap: RoutesMap = {
       }
     ]
   },
+  '/roles/\\S+': {
```
Comment (insignias) [inline workbench-core/example/app/src/configs/staticRouteConfig.ts:110]: Will this match special characters as well ?

**Comment by insignias** [comment]
@bhudnell Has this been deployed and tested ?

File: workbench-core/example/app/src/configs/staticRouteConfig.ts (hunk 1 above)
Comment (bhudnell) [reply workbench-core/example/app/src/configs/staticRouteConfig.ts:110]: It matches all non-whitespace characters. I'm not 100% happy with how this is implemented, so if you have a suggestion on how to better organize the route I'd love to hear it!

File: workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationInputs/getUserGroups.ts [hunk 2]
```diff
@@ -23,7 +23,12 @@ export interface GetUserGroupsRequest {
  */
 export interface GetUserGroupsResponse {
   /**
-   * A list of group ids associated to the user
+   * The data object returned
    */
-  groupIds: string[];
+  data: {
```
[the remark this answers is not in the document]
Comment (bhudnell) [reply workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationInputs/getUserGroups.ts:28]: It was a discussion on another PR:
```
Separation of concerns and consistency.

We work towards a point where all of our response objects have the values you asked for in data.

then the stuff you didn’t ask for like pageToken or stuff you want secondary like metadata, performance data, etc, is separated from the data.
```

**Comment by bhudnell** [comment]
> @bhudnell Has this been deployed and tested ?

Yep! All integ tests pass in my account's instance

File: workbench-core/example/app/src/configs/staticRouteConfig.ts (hunk 1 above)
Comment (insignias) [reply workbench-core/example/app/src/configs/staticRouteConfig.ts:110]: is this for matching roleNames ? and roleNames will be just alphabets then we can at least restrict it to match alphabets only [A-Za-z]+ ?

Comment (bhudnell) [reply workbench-core/example/app/src/configs/staticRouteConfig.ts:110]: These are the requirements for Cognito groups:
```
    The name of the group. Must be unique.

    Type: String

    Length Constraints: Minimum length of 1. Maximum length of 128.

    Pattern: [\p{L}\p{M}\p{S}\p{N}\p{P}]+

    Required: Yes
```
The regex I believe is equivalent to what I have, just not the length

Comment (insignias) [reply workbench-core/example/app/src/configs/staticRouteConfig.ts:110]: \S{1,128} ?

**Review by insignias — APPROVED** [review]

File: workbench-core/example/app/src/configs/staticRouteConfig.ts (hunk 1 above)
Comment (bhudnell) [reply workbench-core/example/app/src/configs/staticRouteConfig.ts:110]: I'll change in a followup PR

**Pull request merged**